ext/webidl/00_webidl.js

```diff
@@ -919,6 +921,125 @@ function createSequenceConverter(converter) {
   };
 }
 
+function isIterator(obj) {
+  if (obj[SymbolAsyncIterator] === undefined) {
+    if (obj[SymbolIterator] === undefined) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+const AsyncIterable = Symbol("[[asyncIterable]]");
+
+function createAsyncIterableConverter(converter) {
+  return function (
+    V,
+    prefix = undefined,
+    context = undefined,
+    opts = { __proto__: null },
+  ) {
+    if (type(V) !== "Object") {
+      throw makeException(
+        TypeError,
+        "can not be converted to async iterable.",
+        prefix,
+        context,
+      );
+    }
+
+    let isAsync = true;
+    let method = V[SymbolAsyncIterator];
+    if (method === undefined) {
+      method = V[SymbolIterator];
+
+      if (method === undefined) {
+        throw new TypeError("No iterator found.");
+      }
+
+      isAsync = true;
+    }
+
+    return {
+      value: V,
+      [AsyncIterable]: AsyncIterable,
+      open() {
+        const iter = FunctionPrototypeCall(method, V);
+        if (type(iter) !== "Object") {
+          throw makeException(
+            TypeError,
+            "invalid iterator.",
+            prefix,
+            context,
+          );
```

Comment:
```suggestion
          throw new TypeError(`${context} could not be iterated because iterator method did not return object, but ${type(iter)}.`);
```